pkgs/applications/terminal-emulators/kitty/default.nix

```diff
@@ -180,9 +180,10 @@ buildPythonApplication rec {
     '';
 
   installPhase = ''
+    set -o nounset
```

Comment:
```
/nix/store/zx2jzw5j0lxigaq34wmyfkcia3n0cvxi-install-shell-files/nix-support/setup-hook: line 100: $1: unbound variable
```

I don't think this is worth setting like this because if someone changes one of the hooks but doesn't build kitty then the breakage will go undiscovered. Until the channels are updates (like what seems to have happened here)

a better place to start adding these would be `pkgs/test/`

example
```
diff --git a/pkgs/test/install-shell-files/default.nix b/pkgs/test/install-shell-files/default.nix
index aef5acc1de6b..27363175cfc3 100644
--- a/pkgs/test/install-shell-files/default.nix
+++ b/pkgs/test/install-shell-files/default.nix
@@ -5,7 +5,7 @@ let
     runCommandLocal "install-shell-files--${name}" ({
       nativeBuildInputs = [ installShellFiles ];
       meta.platforms = lib.platforms.all;
-    } // env) buildCommand;
+    } // env) ("set -o nounset\n" + buildCommand);
 in
 
 recurseIntoAttrs {
```

```diff
$ nix build ".#tests.install-shell-files.install-completion"
this derivation will be built:
  /nix/store/bblpr8v64b153v0p842hv7fp2dvppy59-install-shell-files--install-completion.drv
building '/nix/store/bblpr8v64b153v0p842hv7fp2dvppy59-install-shell-files--install-completion.drv'...
install-shell-files> /nix/store/zx2jzw5j0lxigaq34wmyfkcia3n0cvxi-install-shell-files/nix-support/setup-hook: line 100: $1: unbound variable
```

Reply:
I've removed the `set -o nounset` line - adding it to a more central test file should probably go into a separate PR.

Reply:
Changing the tests won't cause many rebuilds but fixing the issue will of course.

`set -eu`,`set -o pipefail`, and `shopt -s inherit_errexit` are already set `setup.sh` but for them to be set during the build i guess they should be set in `default-builder.sh`